test/Elastic.Apm.Tests/BasicAgentTests.cs

```diff
@@ -87,6 +87,38 @@ public async Task PayloadSentWithBearerToken()
 			authHeader.Parameter.Should().Be(secretToken);
 		}
 
+		[Fact]
+		public async Task PayloadSentWithProperUserAgent()
+		{
+			var isRequestFinished = new TaskCompletionSource<object>();
+
+			HttpHeaderValueCollection<ProductInfoHeaderValue> userAgentHeader = null;
+			var handler = new MockHttpMessageHandler((r, c) =>
+			{
+				userAgentHeader = r.Headers.UserAgent;
+				isRequestFinished.SetResult(null);
+				return Task.FromResult(new HttpResponseMessage(HttpStatusCode.OK));
+			});
+
+			var logger = ConsoleLogger.Instance;
```

Comment:
A small nitpick - we have `NoopLogger` that we use in tests that don't care about what SUT writes to logs.